code/game/machinery/modular_shield.dm

```diff
@@ -178,7 +178,7 @@
 
 		if(exterior_only)
 			for(var/turf/target_tile as anything in list_of_turfs)
-				if(!(isfloorturf(target_tile)) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+				if(!(isfloorturf(target_tile)) && isopenturf(target_tile) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
```

Comment:
Can you change this to a few early continues instead? It's getting pretty cluttered at this point
```dm
if(isfloorturf(target_tile))
    continue
if(!isopenturf(target_tile))
    continue
if(locate(/obj/structure/emergency_shield/modular) in target_tile)
    continue
...
```